trie_query/src/main.rs

```diff
@@ -1,16 +1,51 @@
 use std::path::PathBuf;
 
 use clap::Parser;
-use common::read_input_from_file;
+use common::{read_input_from_file, TRIE_PATH_DOC_STR};
 use mpt_trie::{
     debug_tools::query::{get_path_from_query, DebugQueryParamsBuilder},
     nibbles::Nibbles,
 };
 
+const QUERY_CONFIG_STR: &str = "Query Config";
+
+/// Displays a "trace" of a key query for a given key, where the trace contains the nodes encountered.
 #[derive(Debug, Parser)]
+#[command(after_help = TRIE_PATH_DOC_STR)]
 struct ProgArgs {
+    /// Path to the trie to query against.
     trie_path: PathBuf,
+
+    /// The key to query in hex format (with or without the "0x").
     key: Nibbles,
+
+    /// Config for the query output.
+    #[command(flatten)]
+    config: QueryConfig,
+}
+
+#[derive(Parser, Debug)]
+struct QueryConfig {
+    /// Include the key piece (if the node contains a piece of the key) contained in each node.
```

Comment:
nit (otherwise sound redundant)
```suggestion
    /// Include the key piece (if the node contains a piece of the key) for each node.
```